File: src/main/java/bad/robot/radiate/monitor/MultipleBuildsDemo.java
package bad.robot.radiate.monitor;

import java.util.List;
import java.util.function.IntFunction;
import java.util.function.Supplier;
import java.util.stream.Stream;

import static java.util.stream.Collectors.toList;
import static java.util.stream.IntStream.range;
import static java.util.stream.Stream.*;

@Deprecated
class MultipleBuildsDemo extends ThreadSafeObservable implements MonitoringTasksFactory {

    @Override
    public List<MonitoringTask> create() {
        return generate(RandomStatus::new).limit(4 * 4).collect(toList());
    }
}

File: src/main/scala/bad/robot/radiate/monitor/MultipleBuildsDemo.scala
package bad.robot.radiate.monitor

import bad.robot.radiate.Error
import scalaz.\/
import scalaz.syntax.either._

class MultipleBuildsDemo extends ThreadSafeObservable with MonitoringTasksFactory {
  def create: Error \/ List[MonitoringTask] = {
    Stream.continually(new RandomStatus).take(4 * 4).toList.right
  }
}